### ISSUE TITLE
[Bug] Parse error when executing against psm1 with class inside

### ISSUE THREAD

**Issue by forensicsguy20012004** [MathieuBuisson/PSCodeHealth#6]
## Expected Behavior
I should be able to execute Invoke-PSCodeHealth against a psm1 file that contains both functions and classes.

## Current Behavior

> Invoke-ScriptAnalyzer : Parse error in script definition:  An expression was expected after '(' at line 1 >column 9.
>At C:\Program Files\WindowsPowerShell\Modules\PSCodeHealth\0.2.1\Private\Metrics\Get->FunctionScriptAnalyzerResult.ps1:30 char:16
>+ ...  $Results = Invoke-ScriptAnalyzer -ScriptDefinition $FunctionDefiniti ...
>    + CategoryInfo          : ParserError: (ExpectedExpression:String) [Invoke-ScriptAnalyzer], >ParseException
>    + FullyQualifiedErrorId : Parse error in script definition:  An expression was expected after '(' at line 1 >column 9.,Microsoft.Windows.PowerShell.ScriptAnalyzer.Commands.InvokeScriptAnalyzerCommand

## Possible Solution
I would have to dig into the code but my thought would be to fix the error dealing with '('

## Steps to Reproduce (for bugs)
1. create a module 
2. inside the module create a class 
3. execute Invoke-ScriptAnalyzer -Path ....\myModuleFile.psm1

## Context
I wanted to show off your module to my co-workers but it did not execute against the module file when there was a class in it. Where I work we do all of our work in modules and classes.

## Your Environment
* Module version used: 0.2.1
* Operating System and PowerShell version: Windows 10 [1609]; Powershell 5.1

**Comment by MathieuBuisson** [comment]
Hi @forensicsguy20012004 ,  

I tried to reproduce the problem in all combinations of the following conditions :  

  - With PSScriptAnalyzer 1.16.1
  - With PSScriptAnalyzer 1.11.1
  - Module containing only 1 class
  - Module containing 1 class and 1 function

I was unable to get this error.  

Can you provide the version of PSScriptAnalyzer you are using, please :  
```powershell
(Get-Module PSScriptAnalyzer).Version
```

If `PSScriptAnalyzer` throws this error when analyzing your file even outside of `PSCodeHealth`, then I suspect the error has to do with the content of the file.  
In this case, can you provide the full content of a module file triggering this problem, please ?  

Thank you.

**Comment by SeeminglyScience** [comment]
@MathieuBuisson Did the class you tested with contain a method?

Here's a quick way to reproduce

```powershell
'class Repo {
    [void] Test() { }
}' | Out-File ./testmodule.psm1
Invoke-PSCodeHealth ./testmodule.psm1
```

The `FunctionMemberAst` that makes up class methods has a `FunctionDefinitionAst` inside of it, but it can't be parsed without the rest of the class.  When you're walking the AST for functions, you can exclude methods with something like this:

```powershell
$functionAst.Parent -isnot [System.Management.Automation.Language.FunctionMemberAst]
```

Great work on the module by the way, very cool!

**Comment by MathieuBuisson** [comment]
Hi @SeeminglyScience ,  

You're absolutely right, we hit the problem as soon as we have a method in the class.  

And yes, I'm going to filter out any `AST` which has a Parent of the type `[System.Management.Automation.Language.FunctionMemberAst]`, just like you indicated.  

Thanks a lot for the help !

**Comment by MathieuBuisson** [comment]
Also @forensicsguy20012004 ,  

`PSCodeHealth` doesn't support classes, at the moment.  
The main reason is that unfortunately, the AST-based parsing API doesn't provide much in the way of handling classes.  
The API I'm talking about here is **System.Management.Automation.Language** in the .NET Framework 4.x.  
This may be a bit different in .NET Core, but I haven't looked at it yet.  

I feel that writing a custom parser for PowerShell classes would add quite a bit of complexity.  
So for the time being, I'd rather wait for a proper parsing API for classes and classes members that `PSCodeHealth` could leverage.  

That said, I have a fix/workaround for the error you raised.  
This is excluding class methods when extracting function definitions from the analyzed file(s).  
That way, we don't ask `PSScriptAnalyzer` to parse methods anymore, so it doesn't get confused.

**Comment by MathieuBuisson** [comment]
This fix is included in version 0.2.6 of `PSCodeHealth`.
This version is available [in the PSGallery](https://www.powershellgallery.com/packages/PSCodeHealth/0.2.6) .

**Comment by forensicsguy20012004** [comment]
Awesome. Yes my class had many methods. Plus the module had regular functions as well
